python/ray/data/context.py

```diff
@@ -171,6 +175,8 @@ def __init__(
         self.enable_auto_log_stats = enable_auto_log_stats
         self.trace_allocations = trace_allocations
         self.optimizer_enabled = optimizer_enabled
+        # TODO: expose execution options in Dataset public APIs.
```

Comment:
It'd be better to create a tracking issue

Reply:
Added to https://github.com/ray-project/ray/issues/31797